demos/gesture_recognition_demo/cpp_gapi/include/custom_kernels.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <opencv2/gapi.hpp>
+#include <opencv2/gapi/cpu/gcpukernel.hpp>
+#include <opencv2/gapi/gkernel.hpp>
+
+#include "tracker.hpp"
+
+namespace custom {
+G_API_OP(GetFastFrame,
+         <cv::GMat(cv::GMat, cv::Size)>, "custom.get_fast_frame") {
+    static cv::GMatDesc outMeta(const cv::GMatDesc &in,
+                                const cv::Size& frame_size) {
+        return cv::GMatDesc{CV_8U, 3, frame_size};
+    }
+};
+
+G_API_OP(BoundingBoxExtract,
+         <cv::GArray<TrackedObject>(cv::GMat,
+                                    cv::GMat,
+                                    cv::Scalar)>,
+         "custom.bb_extract") {
+    static cv::GArrayDesc outMeta(const cv::GMatDesc &in,
+                                  const cv::GMatDesc&,
+                                  const cv::Scalar) {
+        return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(PersonTrack,
+         <cv::GArray<TrackedObject>(cv::GMat,
+                                    cv::GArray<TrackedObject>)>,
+         "custom.track") {
+    static cv::GArrayDesc outMeta(const cv::GMatDesc &in,
+                                  const cv::GArrayDesc&) {
+        return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(ConstructClip,
+         <cv::GArray<cv::GMat>(const cv::GMat,
+                               const cv::GArray<TrackedObject>,
+                               cv::GOpaque<std::shared_ptr<int>>,
```

Comment:
what's the reason to make it `shared_ptr` -? should this be a pointer to atomic instead?